src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -19,6 +20,121 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class Binder
     {
+#nullable enable
+        /// <summary>
+        /// This is an argument that needs to be considered for MAMM rules. It's at least an 
+        /// output argument of the function and very often also an input. Essentially this 
+        /// represents all the in, ref and out arguments.
+        /// </summary>
+        internal readonly struct MixableArgument
```

Comment:
Let's consider whether we can name this in a way that makes it clear we are talking about a "destination" of a mixing scenario.

Reply:
How about `MixableDestination`?